xarray/backends/netCDF4_.py

```diff
@@ -343,11 +343,22 @@ def set_dimension(self, name, length, is_unlimited=False):
             dim_length = length if not is_unlimited else None
             self.ds.createDimension(name, size=dim_length)
 
+    def is_list_of_strings(self, value):
+        if (np.asarray(value).dtype.kind in ['U', 'S'] and
+                np.asarray(value).size > 1):
+            return True
+        else:
+            return False
+
     def set_attribute(self, key, value):
         with self.ensure_open(autoclose=False):
             if self.format != 'NETCDF4':
                 value = encode_nc3_attr_value(value)
-            self.ds.setncattr(key, value)
+            if self.is_list_of_strings(value) and (self.format == 'NETCDF4'):
+                # encode as NC_STRING if attr is list of strings and NETCDF4
+                self.ds.setncattr_string(key, value)
```

Comment:
Can we adjust this to raise a sensible error message if the user has an old version of netCDF4-python installed which doesn't have this method?